crates/blockifier/src/execution/syscalls/syscall_tests.rs

```diff
@@ -191,92 +191,120 @@ fn test_call_contract(
+            data: EventData(DATA.to_vec()),
+        };
+        assert_eq!(
+            call_info.execution,
+            CallExecution {
+                events: vec![OrderedEvent { order: 0, event }],
+                gas_consumed: expected_gas,
+                ..Default::default()
+            }
+        );
+    }
 
-    // Negative flow, the data length exceeds the limit.
-    let max_event_data_length = versioned_constants.tx_event_limits.max_data_length;
-    let data_too_long = vec![stark_felt!(2_u16); max_event_data_length + 1];
-    let error = emit_events(test_contract, &n_emitted_events, &keys, &data_too_long).unwrap_err();
-    let expected_error = EmitEventError::ExceedsMaxDataLength {
-        data_length: max_event_data_length + 1,
-        max_data_length: max_event_data_length,
-    };
-    assert!(error.to_string().contains(format!("{}", expected_error).as_str()));
-
-    // Negative flow, the keys length exceeds the limit.
-    let max_event_keys_length = versioned_constants.tx_event_limits.max_keys_length;
-    let keys_too_long = vec![stark_felt!(1_u16); max_event_keys_length + 1];
-    let error = emit_events(test_contract, &n_emitted_events, &keys_too_long, &data).unwrap_err();
-    let expected_error = EmitEventError::ExceedsMaxKeysLength {
-        keys_length: max_event_keys_length + 1,
-        max_keys_length: max_event_keys_length,
-    };
-    assert!(error.to_string().contains(format!("{}", expected_error).as_str()));
-
-    // Negative flow, the number of events exceeds the limit.
-    let max_n_emitted_events = versioned_constants.tx_event_limits.max_n_emitted_events;
-    let n_emitted_events_too_big = vec![stark_felt!(
-        u16::try_from(max_n_emitted_events + 1).expect("Failed to convert usize to u16.")
-    )];
-    let error = emit_events(test_contract, &n_emitted_events_too_big, &keys, &data).unwrap_err();
-    let expected_error = EmitEventError::ExceedsMaxNumberOfEmittedEvents {
-        n_emitted_events: max_n_emitted_events + 1,
-        max_n_emitted_events,
-    };
-    assert!(error.to_string().contains(format!("{}", expected_error).as_str()));
-}
+    #[test_case(FeatureContract::SierraTestContract; "Native")]
+    #[test_case(FeatureContract::TestContract(CairoVersion::Cairo1); "VM")]
+    fn data_length_exceeds_limit(test_contract: FeatureContract) {
+        let versioned_constants = VersionedConstants::create_for_testing();
+
+        let max_event_data_length = versioned_constants.tx_event_limits.max_data_length;
+        let data_too_long = vec![stark_felt!(2_u16); max_event_data_length + 1];
+        let error =
+            emit_events(test_contract, &N_EMITTED_EVENTS, &KEYS, &data_too_long).unwrap_err();
+        let expected_error = EmitEventError::ExceedsMaxDataLength {
+            data_length: max_event_data_length + 1,
+            max_data_length: max_event_data_length,
+        };
+        assert!(error.to_string().contains(format!("{}", expected_error).as_str()));
```

Comment:
```suggestion
        assert!(error.to_string().contains(&expected_error.to_string()));
```
can we do it in this way?

Reply:
Actually, I was mantaining Starkware's format. Currently I cannot think of any reason for them to keep it like that. I think it might be legacy code. I'll update it to your suggestion. Also let's try to keep it consistent with the rest of the syscalls in future PRs